developmentseed/scoreboard
# add/edit badges with date-specific criteria

Changes include:

- UI for adding & editing badges with date-specific requirements
- A few small changes on the server side to get things working
- Minimal refactoring of badge-add.js page so that we're not switching operations between objects and arrays which made things a little easier for working with the date operations and reduced differences between badge-add.js and badge-edit.js

Example of the input fields for date-specific metrics:

<img width="279" alt="screen shot 2019-01-10 at 2 05 40 pm" src="https://user-images.githubusercontent.com/164214/51000090-e2188b80-14e0-11e9-8801-673fa0bc76d6.png">

So far have tested creating and updating badges with a bunch of different combinations via the UI.

Steps for testing included the usual:

- reset the db `npm run rollback && npm run migrate && npm run seed`
- reinstalled modules `npm i`
- started the server `npm run dev`
- added myself as admin `scoreboard create-admin-user -u sethvincent`
- added and edited badges via the admin ui
- checked out the api endpoint to verify the data is right

Closes #218
